src/components/common/Typography/Form/CollapsibleArrayTemplate/CollapsibleArrayTemplate.jsx

```diff
@@ -17,71 +18,97 @@ const renderLabel = (label, displayLabel) => {
 /**
  * This is collapsible array template for array field.
  */
-const CollapsibleArrayTemplate = (props) => {
-  const [indexToOpen, changeIndex] = useState(undefined);
-  const {
-    id,
-    classNames,
-    label,
-    displayLabel,
-    errors,
-    items,
-    uiSchema,
-    canAdd,
-    title,
-  } = props;
-
-  const showErrors = uiSchema && uiSchema['ui:options'] ? !!uiSchema['ui:options'].showErrors : true;
-
-  // For rendering the errors to the field.
-  const renderErrors = () => {
-    if (showErrors && id !== 'root') {
-      return (
-        <div className="form-field-errors">
-          {errors}
-        </div>
-      );
-    }
-    return null;
-  };
+class CollapsibleArrayTemplate extends React.Component {
+  constructor(props) {
+    super(props);
 
-  const renderTitle = () => {
-    if (title && title.length) {
-      return <p className="collapsible-array-template__form-expander-heading">{title}</p>;
-    }
-    return null;
-  };
-
-  const onClickEdit = (index, isClicked) => {
-    if (isClicked) {
-      changeIndex(index);
-    } else {
-      changeIndex(undefined);
+    this.state = {
+      indexToOpen: undefined,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    // This code will automatically opens the newly add item
+    if ((prevProps.formData).length < (this.props.formData).length) {
+      if ((prevProps.formData).length === 1 && (isEmpty(prevProps.formData[0]) || !prevProps.formData[0].name)) {
```

Comment:
Please simplify this if condition
```suggestion
      if ((prevProps.formData).length) {
```
(isEmpty(prevProps.formData[0]) || !prevProps.formData[0].name)
If formData is empty OR formData does not have name ? I didn't get it.

Reply:
There are 2 conditions one where the object is empty & one where we get name empty, I want to  use both the conditions.

Reply:
If the formData object is empty. Then how can it may have name inside?  Then it should be:
```suggestion
      if ((prevProps.formData).length &&  !prevProps.formData[0].name) {
```

Reply:
Its is or condition either object is empty or object have name that is empty.